[//]: # "File generated from a template. Do not edit this file directly."

# cred-class-field-display-name-missing-api

`displayName` field in credential class must be end with `API`.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/recommended` config.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

ğŸ”§ Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

âŒ Example of **incorrect** code:

```js
class TestApi implements ICredentialType {
  name = "testApi";
  displayName = "Test";
  documentationUrl = "test";
}
```

âœ… Example of **correct** code:

```js
class TestApi implements ICredentialType {
  name = "testApi";
  displayName = "Test API";
  documentationUrl = "test";
}
```

## Links

- [Rule source](../../lib/rules/cred-class-field-display-name-missing-api.ts)
- [Test source](../../tests/cred-class-field-display-name-missing-api.test.ts)
