[//]: # "File generated from a template. Do not edit this file directly."

# cred-class-field-name-uppercase-first-char

First char in `name` in credential class must be lowercase.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/credentials` config.

ğŸ”§ Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

âŒ Example of **incorrect** code:

```js
class TestApi implements ICredentialType {
  name = "MyTestApi";
  displayName = "My Test API";
  documentationUrl = "myTest";
}
```

âœ… Example of **correct** code:

```js
class MyTestApi implements ICredentialType {
  name = "myTestApi";
  displayName = "My Test API";
  documentationUrl = "myTest";
}
```

## Links

- [Rule source](../../lib/rules/cred-class-field-name-uppercase-first-char.ts)
- [Test source](../../tests/cred-class-field-name-uppercase-first-char.test.ts)
