[//]: # "File generated from a template. Do not edit this file directly."

# cred-class-field-authenticate-type-assertion

In a credential class, the field `authenticate` must be typed `IAuthenticateGeneric`

üìã This rule is part of the `plugin:n8n-nodes-base/credentials` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
class TestApi implements ICredentialType {
    name = 'myTestApi';
    displayName = 'My Test API';
    documentationUrl = 'myTest';
    authenticate = {
        type: 'generic',
        properties: {
            headers: {
                Authorization: '=Token {{$credentials.token}}',
            },
        },
    } as IAuthenticateGeneric;
}
```

‚úÖ Example of **correct** code:

```js
class TestApi implements ICredentialType {
	name = "myTestApi";
	displayName = "My Test API";
	documentationUrl = "myTest";
	authenticate: IAuthenticateGeneric = {
		type: "generic",
		properties: {
			headers: {
				Authorization: "=Token {{$credentials.token}}",
			},
		},
	};
}
```

## Links

- [Rule source](../../lib/rules/cred-class-field-authenticate-type-assertion.ts)
- [Test source](../../tests/cred-class-field-authenticate-type-assertion.test.ts)
