[//]: # "File generated from a template. Do not edit this file directly."

# cred-class-field-display-name-miscased

`displayName` field in credential class must be title cased, except for `n8n API` and `E-goi API`

üìã This rule is part of the `plugin:n8n-nodes-base/credentials` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
class TestApi implements ICredentialType {
    name = 'myTestApi';
    displayName = 'My test API';
    documentationUrl = 'myTest';
}
```

‚úÖ Example of **correct** code:

```js
class MyTestApi implements ICredentialType {
    name = 'myTestApi';
    displayName = 'My Test API';
    documentationUrl = 'myTest';
}

class MyTestApi implements ICredentialType {
    name = 'myTestApi';
    displayName = 'n8n API';
    documentationUrl = 'myTest';
}
```

## Links

- [Rule source](../../lib/rules/cred-class-field-display-name-miscased.ts)
- [Test source](../../tests/cred-class-field-display-name-miscased.test.ts)
