[//]: # "File generated from a template. Do not edit this file directly."

# cred-class-field-name-missing-oauth2

`name` field in credential class must mention `OAuth2` if the credential is OAuth2.

üìã This rule is part of the `plugin:n8n-nodes-base/credentials` config.

## Examples

‚ùå Example of **incorrect** code:

```js
class MyTestOAuth2Api implements ICredentialType {
  name = "myTestApi";
  displayName = "My Test OAuth2 API";
  documentationUrl = "myTest";
  extends = ["oAuth2Api"];
}
```

‚úÖ Example of **correct** code:

```js
class MyTestOAuth2Api implements ICredentialType {
  name = "myTestOAuth2Api";
  displayName = "My Test OAuth2 API";
  documentationUrl = "myTest";
  extends = ["oAuth2Api"];
}
```

## Links

- [Rule source](../../lib/rules/cred-class-field-name-missing-oauth2.ts)
- [Test source](../../tests/cred-class-field-name-missing-oauth2.test.ts)
