[//]: # "File generated from a template. Do not edit this file directly."

# cred-class-field-documentation-url-not-http-url

`documentationUrl` field in credential class must be an HTTP URL. Only applicable to community credentials.

üìã This rule is part of the `plugin:n8n-nodes-base/credentials` config.

## Examples

‚ùå Example of **incorrect** code:

```js
class TestApi implements ICredentialType {
  name = "myTestApi";
  displayName = "My Test API";
  documentationUrl = "lowercase";
}

class TestApi implements ICredentialType {
  name = "myTestApi";
  displayName = "My Test API";
  documentationUrl = "camelCase";
}

class TestApi implements ICredentialType {
  name = "myTestApi";
  displayName = "My Test API";
  documentationUrl = "PascalCase";
}

class TestApi implements ICredentialType {
  name = "myTestApi";
  displayName = "My Test API";
  documentationUrl = "Sentence case";
}
```

‚úÖ Example of **correct** code:

```js
class MyTestApi implements ICredentialType {
  name = "myTestApi";
  displayName = "My Test API";
  documentationUrl = "http://example.com/docs/auth";
}

class MyTestApi implements ICredentialType {
  name = "myTestApi";
  displayName = "My Test API";
  documentationUrl = "https://example.com/docs/auth";
}
```

## Links

- [Rule source](../../lib/rules/cred-class-field-documentation-url-not-http-url.ts)
- [Test source](../../tests/cred-class-field-documentation-url-not-http-url.test.ts)
