[//]: # "File generated from a template. Do not edit this file directly."

# cred-class-field-documentation-url-miscased

`documentationUrl` field in credential class must be camel cased.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/recommended` config.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

ğŸ”§ Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

âŒ Example of **incorrect** code:

```js
class TestApi implements ICredentialType {
  name = "myTestApi";
  displayName = "My Test API";
  documentationUrl = "MyTest";
}
```

âœ… Example of **correct** code:

```js
class MyTestApi implements ICredentialType {
  name = "myTestApi";
  displayName = "My Test API";
  documentationUrl = "myTest";
}
```

## Links

- [Rule source](../../lib/rules/cred-class-field-documentation-url-miscased.ts)
- [Test source](../../tests/cred-class-field-documentation-url-miscased.test.ts)
