[//]: # "File generated from a template. Do not edit this file directly."

# cred-class-field-name-unsuffixed

`name` field in credential class must be suffixed with `-Api`.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/credentials` config.

ğŸ”§ Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

âŒ Example of **incorrect** code:

```js
class TestApi implements ICredentialType {
    name = 'test';
    displayName = 'Test API';
    documentationUrl = 'test';
}
```

âœ… Example of **correct** code:

```js
class TestApi implements ICredentialType {
    name = 'testApi';
    displayName = 'Test API';
    documentationUrl = 'test';
}
```

## Links

- [Rule source](../../lib/rules/cred-class-field-name-unsuffixed.ts)
- [Test source](../../tests/cred-class-field-name-unsuffixed.test.ts)
