[//]: # "File generated from a template. Do not edit this file directly."

# node-execute-block-operation-missing-plural-pairing

Every `getAll` operation in the `execute()` method in a node must implement plural pairing.

üìã This rule is part of the `plugin:n8n-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
class TestNode {
    async execute() {
        const items = this.getInputData();
        const returnData: INodeExecutionData[] = [];

        const resource = this.getNodeParameter('resource', 0) as string;
        const operation = this.getNodeParameter('operation', 0) as string;

        let responseData;

        for (let i = 0; i < items.length; i++) {

            try {

                if (resource === 'ticket') {

                    if (operation === 'create') {

                        responseData = await serviceApiRequest.call(this, 'POST', '/tickets', body);

                        returnData.push({
                            json: responseData,
                            pairedItem: {
                                item: i,
                            },
                        });

                    } else if (operation === 'getAll') {

                        responseData = await serviceApiRequest.call(this, 'GET', '/tickets', body);

                    }

                }

            } catch (error) {
                if (this.continueOnFail()) {
                    // ...
                }
            }
        }

        return [returnData];
    }
}
```

‚úÖ Example of **correct** code:

```js
class TestNode {
    async execute() {
        const items = this.getInputData();
        const returnData: INodeExecutionData[] = [];

        const resource = this.getNodeParameter('resource', 0) as string;
        const operation = this.getNodeParameter('operation', 0) as string;

        let responseData;

        for (let i = 0; i < items.length; i++) {

            try {

                if (resource === 'ticket') {

                    if (operation === 'create') {

                        responseData = await serviceApiRequest.call(this, 'POST', '/tickets', body);

                        returnData.push({
                            json: responseData,
                            pairedItem: {
                                item: i,
                            },
                        });

                    } else if (operation === 'getAll') {

                        responseData = await serviceApiRequest.call(this, 'GET', '/tickets', body);

                        returnData.push(
                            ...responseData.map((json) => {
                                return {
                                    json,
                                    pairedItem: {
                                        item: i,
                                    },
                                };
                            })
                        );

                    }

                }

            } catch (error) {
                if (this.continueOnFail()) {
                    // ...
                }
            }
        }

        return [returnData];
    }
}
```

## Links

- [Rule source](../../lib/rules/node-execute-block-operation-missing-plural-pairing.ts)
- [Test source](../../tests/node-execute-block-operation-missing-plural-pairing.test.ts)
