[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-unencoded-angle-brackets

`description` in node parameter must encode angle brackets for them to render. Applicable by extension to `description` in option in options-type and multi-options-type node parameter.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/recommended` config.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

ğŸ”§ Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

âŒ Example of **incorrect** code:

```js
const test = {
  displayName: "Test",
  name: "test",
  type: "string",
  default: "",
  description: "This is a <value>",
};

const test = {
  name: "Test",
  value: "test",
  description: "This is a <value>",
};
```

âœ… Example of **correct** code:

```js
const test = {
  displayName: "Test",
  name: "test",
  type: "string",
  default: "",
  description: "This is a <b>sentence</b>",
};

const test = {
  name: "Test",
  value: "test",
  description: "This is a <b>sentence</b>",
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-unencoded-angle-brackets.ts)
- [Test source](../../tests/node-param-description-unencoded-angle-brackets.test.ts)
