[//]: # "File generated from a template. Do not edit this file directly."

# node-class-description-name-unsuffixed-trigger-node

`name` in node class description for trigger node must be suffixed with `-Trigger`.

üìã This rule is part of the `plugin:n8n-nodes-base/recommended` config.

üìã This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
class TestNode {
  description = {
    displayName: "TestTrigger",
    name: "test",
    icon: "file:test.svg",
    group: ["transform"],
    version: 1,
    subtitle: '={{ $parameter["operation"] + ": " + $parameter["resource"] }}',
    description: "This is a sentence",
    defaults: {
      name: "Test",
    },
    inputs: ["main"],
    outputs: ["main"],
  };
}
```

‚úÖ Example of **correct** code:

```js
class TestNode {
  description = {
    displayName: "TestTrigger",
    name: "testTrigger",
    icon: "file:test.svg",
    group: ["transform"],
    version: 1,
    subtitle: '={{ $parameter["operation"] + ": " + $parameter["resource"] }}',
    description: "This is a sentence",
    defaults: {
      name: "Test",
    },
    inputs: ["main"],
    outputs: ["main"],
  };
}
```

## Links

- [Rule source](../../lib/rules/node-class-description-name-unsuffixed-trigger-node.ts)
- [Test source](../../tests/node-class-description-name-unsuffixed-trigger-node.test.ts)
