[//]: # "File generated from a template. Do not edit this file directly."

# node-param-type-options-max-value-present

`maxValue` in `typeOptions` in Limit node parameter is deprecated and must not be present.

üìã This rule is part of the `plugin:n8n-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	displayName: "Limit",
	name: "limit",
	default: 50,
	type: "number",
	typeOptions: {
		minValue: 1,
		maxValue: 1,
	},
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	displayName: "Limit",
	name: "limit",
	default: 50,
	type: "number",
	typeOptions: {
		minValue: 1,
	},
};
```

## Links

- [Rule source](../../lib/rules/node-param-type-options-max-value-present.ts)
- [Test source](../../tests/node-param-type-options-max-value-present.test.ts)
