[//]: # "File generated from a template. Do not edit this file directly."

# node-param-default-wrong-for-fixed-collection

`default` for fixed-collection-type node parameter must be an object.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/recommended` config.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

ğŸ”§ Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

âŒ Example of **incorrect** code:

```js
const test = {
  displayName: "Test",
  name: "test",
  type: "fixedCollection",
  default: 1,
};
```

âœ… Example of **correct** code:

```js
const test = {
  displayName: "Test",
  name: "test",
  type: "fixedCollection",
  default: {},
};
```

## Links

- [Rule source](../../lib/rules/node-param-default-wrong-for-fixed-collection.ts)
- [Test source](../../tests/node-param-default-wrong-for-fixed-collection.test.ts)
