[//]: # "File generated from a template. Do not edit this file directly."

# node-param-display-name-untrimmed

`displayName` in node parameter or in fixed collection section must be trimmed. Applicable by extension to `name` in options-type or multi-options-type node parameter.

üìã This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
  displayName: "Operation   ",
  name: "operation",
  type: "string",
  default: "",
};

const test = {
  name: "Operation   ",
  value: "operation",
};

const test = {
  displayName: "Test",
  name: "test",
  type: "fixedCollection",
  default: "a",
  options: [
    {
      displayName: "Details   ",
      name: "details",
      values: [
        {
          displayName: "A",
          name: "a",
          type: "string",
          default: "",
        },
      ],
    },
  ],
};
```

‚úÖ Example of **correct** code:

```js
const test = {
  displayName: "Operation",
  name: "operation",
  type: "string",
  default: "",
};

const test = {
  name: "Operation",
  value: "operation",
};
```

## Links

- [Rule source](../../lib/rules/node-param-display-name-untrimmed.ts)
- [Test source](../../tests/node-param-display-name-untrimmed.test.ts)
