[//]: # "File generated from a template. Do not edit this file directly."

# node-param-operation-without-no-data-expression

`noDataExpression` in an Operation node parameter must be present and enabled.

üìã This rule is part of the `plugin:n8n-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	displayName: "Operation",
	name: "operation",
	type: "options",
	noDataExpression: false,
	options: [
		{
			name: "Get",
			value: "get",
		},
	],
	default: "contact",
};

const test = {
	displayName: "Operation",
	name: "operation",
	type: "options",
	options: [
		{
			name: "Get",
			value: "get",
		},
	],
	default: "contact",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	displayName: "Operation",
	name: "operation",
	type: "options",
	noDataExpression: true,
	options: [
		{
			name: "Get",
			value: "get",
		},
	],
	default: "contact",
};
```

## Links

- [Rule source](../../lib/rules/node-param-operation-without-no-data-expression.ts)
- [Test source](../../tests/node-param-operation-without-no-data-expression.test.ts)
