[//]: # "File generated from a template. Do not edit this file directly."

# node-param-color-type-unused

`color`-type must be used for color-related node parameter.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/recommended` config.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/autofixable-unsafe` config.

ğŸ”§ Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

âŒ Example of **incorrect** code:

```js
const test = {
  displayName: "Foreground Colour",
  name: "foregroundColor",
  type: "string",
  default: "#000000",
};
```

âœ… Example of **correct** code:

```js
const test = {
  displayName: "Foreground Colour",
  name: "foregroundColor",
  type: "color",
  default: "#000000",
};
```

## Links

- [Rule source](../../lib/rules/node-param-color-type-unused.ts)
- [Test source](../../tests/node-param-color-type-unused.test.ts)
