[//]: # "File generated from a template. Do not edit this file directly."

# node-param-display-name-miscased-id

`ID` in `displayName` in node parameter must be fully uppercased. Applicable by extension to `name` in options-type or multi-options-type node parameter.

üìã This rule is part of the `plugin:n8n-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
  displayName: "User Id",
  name: "userId",
  type: "string",
  default: "",
};

const test = {
  displayName: "User Ids",
  name: "userId",
  type: "string",
  default: "",
};

const test = {
  displayName: "User id",
  name: "userId",
  type: "string",
  default: "",
};

const test = {
  name: "User id",
  value: "userId",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
  displayName: "User ID",
  name: "userId",
  type: "string",
  default: "",
};

const test = {
  name: "User ID",
  value: "userId",
};

const test = {
  name: "User Identifier",
  value: "userId",
};
```

## Links

- [Rule source](../../lib/rules/node-param-display-name-miscased-id.ts)
- [Test source](../../tests/node-param-display-name-miscased-id.test.ts)
