[//]: # "File generated from a template. Do not edit this file directly."

# node-param-default-wrong-for-multi-options

`default` for a multi-options-type node parameter must be an array.

üìã This rule is part of the `plugin:n8n-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
  displayName: "Test",
  name: "test",
  type: "multiOptions",
  default: "wrong",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
  displayName: "Test",
  name: "test",
  type: "multiOptions",
  default: [],
};
```

## Links

- [Rule source](../../lib/rules/node-param-default-wrong-for-multi-options.ts)
- [Test source](../../tests/node-param-default-wrong-for-multi-options.test.ts)
