[//]: # "File generated from a template. Do not edit this file directly."

# node-param-collection-type-unsorted-items

Items in collection-type node parameter must be alphabetized by `name` if five or more than five.

üìã This rule is part of the `plugin:n8n-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "collection",
	default: {},
	options: [
		{
			displayName: "C",
			name: "c",
			type: "boolean",
			default: true,
		},
		{
			displayName: "A",
			name: "a",
			type: "boolean",
			default: true,
		},
		{
			displayName: "B",
			name: "b",
			type: "boolean",
			default: true,
		},
		{
			displayName: "D",
			name: "d",
			type: "boolean",
			default: true,
		},
		{
			displayName: "E",
			name: "e",
			type: "boolean",
			default: true,
		},
	],
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "collection",
	default: {},
	options: [
		{
			displayName: "C",
			name: "c",
			type: "boolean",
			default: true,
		},
		{
			displayName: "A",
			name: "a",
			type: "boolean",
			default: true,
		},
		{
			displayName: "B",
			name: "b",
			type: "boolean",
			default: true,
		},
		{
			displayName: "D",
			name: "d",
			type: "boolean",
			default: true,
		},
	],
};

const test = {
	displayName: "Test",
	name: "test",
	type: "collection",
	default: {},
	options: [
		{
			displayName: "A",
			name: "a",
			type: "boolean",
			default: true,
		},
		{
			displayName: "B",
			name: "b",
			type: "boolean",
			default: true,
		},
		{
			displayName: "C",
			name: "c",
			type: "boolean",
			default: true,
		},
		{
			displayName: "D",
			name: "d",
			type: "boolean",
			default: true,
		},
	],
};
```

## Links

- [Rule source](../../lib/rules/node-param-collection-type-unsorted-items.ts)
- [Test source](../../tests/node-param-collection-type-unsorted-items.test.ts)
