[//]: # "File generated from a template. Do not edit this file directly."

# node-param-option-description-identical-to-name

`description` in option in options-type node parameter must not be identical to `name`.

üìã This rule is part of the `plugin:n8n-nodes-base/recommended` config.

üìã This rule is part of the `plugin:n8n-nodes-base/non-autofixable` config.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
  name: "Full Name",
  value: "fullName",
  description: "Full Name",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
  name: "Full Name",
  value: "fullName",
  description: "First and last name of the user",
};
```

## Links

- [Rule source](../../lib/rules/node-param-option-description-identical-to-name.ts)
- [Test source](../../tests/node-param-option-description-identical-to-name.test.ts)
