[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-comma-separated-hyphen

The string `comma-separated` in `description` must be hyphenated. Applicable by extension to `description` in option in options-type and multi-options-type node parameter.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/nodes` config.

ğŸ”§ Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

âŒ Example of **incorrect** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	description: "A comma separated list",
};

const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	description: "Comma separated list",
};
```

âœ… Example of **correct** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	description: "A comma-separated list",
};

const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	description: "Comma-separated list",
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-comma-separated-hyphen.ts)
- [Test source](../../tests/node-param-description-comma-separated-hyphen.test.ts)
