[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-wrong-for-simplify

`description` for Simplify node parameter must be `Whether to return a simplified version of the response instead of the raw data`

üìã This rule is part of the `plugin:n8n-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
  displayName: "Simplify",
  name: "simple",
  type: "boolean",
  default: true,
  description: "Wrong",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
  displayName: "Simplify",
  name: "simple",
  type: "boolean",
  default: true,
  description:
    "Whether to return a simplified version of the response instead of the raw data",
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-wrong-for-simplify.ts)
- [Test source](../../tests/node-param-description-wrong-for-simplify.test.ts)
