[//]: # "File generated from a template. Do not edit this file directly."

# node-param-hint-untrimmed

`hint` in node parameter must be trimmed.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/nodes` config.

ğŸ”§ Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

âŒ Example of **incorrect** code:

```js
const test = {
	displayName: "Operation",
	name: "operation",
	type: "string",
	default: "",
	hint: " This is a hint",
};
```

âœ… Example of **correct** code:

```js
const test = {
	displayName: "Operation",
	name: "operation",
	type: "string",
	default: "",
	hint: "This is a hint",
};
```

## Links

- [Rule source](../../lib/rules/node-param-hint-untrimmed.ts)
- [Test source](../../tests/node-param-hint-untrimmed.test.ts)
