[//]: # "File generated from a template. Do not edit this file directly."

# node-class-description-non-core-color-present

`color` in node class description is deprected and must not be present, except for nodes whose icon is a Font Awesome icon - usually core nodes.

üìã This rule is part of the `plugin:n8n-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
class TestNode {
	description = {
		displayName: "TestTrigger",
		name: "testTrigger",
		icon: "file:test.svg",
		group: ["transform"],
		version: 1,
		subtitle: '={{ $parameter["operation"] + ": " + $parameter["resource"] }}',
		description: "This is a sentence",
		defaults: {
			name: "Test",
			color: "#ffffff",
		},
		inputs: ["main"],
		outputs: ["main"],
	};
}
```

‚úÖ Example of **correct** code:

```js
class TestNode {
	description = {
		displayName: "TestTrigger",
		name: "testTrigger",
		icon: "file:test.svg",
		group: ["transform"],
		version: 1,
		subtitle: '={{ $parameter["operation"] + ": " + $parameter["resource"] }}',
		description: "This is a sentence",
		defaults: {
			name: "Test",
		},
		inputs: ["main"],
		outputs: ["main"],
	};
}
```

## Links

- [Rule source](../../lib/rules/node-class-description-non-core-color-present.ts)
- [Test source](../../tests/node-class-description-non-core-color-present.test.ts)
