[//]: # "File generated from a template. Do not edit this file directly."

# node-param-hint-url-missing-protocol

`hint` in node parameter must include protocol e.g. `https://` when containing a URL.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/nodes` config.

ğŸ”§ Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

âŒ Example of **incorrect** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	hint: 'This is a <a href="github.com">link</a>',
};
```

âœ… Example of **correct** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	hint: 'This is a <a href="https://github.com">link</a>',
};

const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	hint: "This is a <a href='https://github.com'>link</a>",
};

const test = {
	name: "Test",
	value: "test",
	hint: 'This is a <a href="https://github.com">link</a>',
};

const test = {
	name: "Test",
	value: "test",
	hint: "This is a <a href='https://github.com'>link</a>",
};
```

## Links

- [Rule source](../../lib/rules/node-param-hint-url-missing-protocol.ts)
- [Test source](../../tests/node-param-hint-url-missing-protocol.test.ts)
