[//]: # "File generated from a template. Do not edit this file directly."

# node-param-array-type-assertion

Array of node parameters must be typed, not type-asserted.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/nodes` config.

ğŸ”§ Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

âŒ Example of **incorrect** code:

```js
const test = [
    {
        displayName: 'Test',
        name: 'test',
        type: 'string',
        default: '',
    },
] as INodeProperties[];
```

âœ… Example of **correct** code:

```js
const test: INodeProperties[] = [
	{
		displayName: "Test",
		name: "test",
		type: "string",
		default: "",
	},
];
```

## Links

- [Rule source](../../lib/rules/node-param-array-type-assertion.ts)
- [Test source](../../tests/node-param-array-type-assertion.test.ts)
