[//]: # "File generated from a template. Do not edit this file directly."

# node-param-display-name-wrong-for-update-fields

`displayName` for Update operation node parameter must be `Update Fields`

üìã This rule is part of the `plugin:n8n-nodes-base/recommended` config.

üìã This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
  displayName: "Wrong",
  name: "updateFields",
  type: "collection",
  placeholder: "Add Field",
  default: {},
  displayOptions: {
    show: {
      resource: ["person"],
      operation: ["update"],
    },
  },
  options: [],
};
```

‚úÖ Example of **correct** code:

```js
const test = {
  displayName: "Update Fields",
  name: "updateFields",
  type: "collection",
  placeholder: "Add Field",
  default: {},
  displayOptions: {
    show: {
      resource: ["person"],
      operation: ["update"],
    },
  },
  options: [],
};

const test = {
  displayName: "User ID",
  name: "userId",
  type: "collection",
  placeholder: "Add Field",
  default: {},
  displayOptions: {
    show: {
      resource: ["user"],
      operation: ["update"],
    },
  },
  options: [],
};
```

## Links

- [Rule source](../../lib/rules/node-param-display-name-wrong-for-update-fields.ts)
- [Test source](../../tests/node-param-display-name-wrong-for-update-fields.test.ts)
