[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-wrong-for-upsert

`description` for Upsert node parameter must be `Create a new record, or update the current one if it already exists (upsert)`

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/recommended` config.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

ğŸ”§ Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

âŒ Example of **incorrect** code:

```js
const test = {
  name: "Upsert",
  value: "upsert",
  description: "Wrong",
};
```

âœ… Example of **correct** code:

```js
const test = {
  name: "Upsert",
  value: "upsert",
  description:
    "Create a new record, or update the current one if it already exists (upsert)",
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-wrong-for-upsert.ts)
- [Test source](../../tests/node-param-description-wrong-for-upsert.test.ts)
