[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-weak

`description` in node parameter must be either useful or omitted. Applicable by extension to `description` in option in options-type and multi-options-type node parameter.

üìã This rule is part of the `plugin:n8n-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	description: "Operation to perform",
};

const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
	description: "The operation to perform",
};

const test = {
	name: "User ID",
	value: "userId",
	description: "Operation to perform",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "string",
	default: "",
};

const test = {
	name: "User ID",
	value: "userId",
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-weak.ts)
- [Test source](../../tests/node-param-description-weak.test.ts)
