[//]: # "File generated from a template. Do not edit this file directly."

# node-param-option-name-wrong-for-upsert

Option `name` for Upsert node parameter must be `Create or Update`.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

ğŸ”§ Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

âŒ Example of **incorrect** code:

```js
const test = {
  name: "Insert or Create",
  value: "upsert",
};
```

âœ… Example of **correct** code:

```js
const test = {
  name: "Create or Update",
  value: "upsert",
};
```

## Links

- [Rule source](../../lib/rules/node-param-option-name-wrong-for-upsert.ts)
- [Test source](../../tests/node-param-option-name-wrong-for-upsert.test.ts)
