[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-unneeded-backticks

`description` in node parameter must not use unneeded backticks. Applicable by extension to `description` in option in options-type and multi-options-type node parameter.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/recommended` config.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

ğŸ”§ Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

âŒ Example of **incorrect** code:

```js
const test = {
  displayName: "Test",
  name: "test",
  type: "string",
  default: "",
  description: `This a sentence`,
};

const test = {
  name: "User ID",
  value: "userId",
  description: `The ID of the user`,
};
```

âœ… Example of **correct** code:

```js
const test = {
  displayName: "Test",
  name: "test",
  type: "string",
  default: "",
  description: "This a sentence",
};

const test = {
  name: "User ID",
  value: "userId",
  description: "The ID of the user",
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-unneeded-backticks.ts)
- [Test source](../../tests/node-param-description-unneeded-backticks.test.ts)
