[//]: # "File generated from a template. Do not edit this file directly."

# node-class-description-missing-subtitle

`subtitle` in node class description must be present.

üìã This rule is part of the `plugin:n8n-nodes-base/recommended` config.

üìã This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
class TestNode {
  description = {
    displayName: "Test",
    name: "test",
    icon: "file:test.svg",
    group: ["transform"],
    version: 1,
    description: "This is a sentence",
    defaults: {
      name: "Test",
    },
    inputs: ["main"],
    outputs: ["main"],
    properties: [
      {
        displayName: "Resource",
      },
      {
        displayName: "Operation",
      },
    ],
  };
}
```

‚úÖ Example of **correct** code:

```js
class TestNode {
  description = {
    displayName: "Test",
    name: "test",
    icon: "file:test.svg",
    group: ["transform"],
    version: 1,
    subtitle: '={{ $parameter["operation"] + ": " + $parameter["resource"] }}',
    description: "This is a sentence",
    defaults: {
      name: "Test",
    },
    inputs: ["main"],
    outputs: ["main"],
    properties: [
      {
        displayName: "Resource",
      },
      {
        displayName: "Operation",
      },
    ],
  };
}

class TestTriggerNode {
  description = {
    displayName: "Test Trigger",
    name: "testTrigger",
    icon: "file:test.svg",
    group: ["transform"],
    version: 1,
    subtitle: '={{ $parameter["operation"] + ": " + $parameter["resource"] }}',
    description: "This is a sentence",
    defaults: {
      name: "Test",
    },
    inputs: ["main"],
    outputs: ["main"],
  };
}
```

## Links

- [Rule source](../../lib/rules/node-class-description-missing-subtitle.ts)
- [Test source](../../tests/node-class-description-missing-subtitle.test.ts)
