[//]: # "File generated from a template. Do not edit this file directly."

# node-param-multi-options-type-unsorted-items

Items in a multi-options-type node parameter must be alphabetized by `name` if five or more than five.

üìã This rule is part of the `plugin:n8n-nodes-base/recommended` config.

üìã This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
  displayName: "Test",
  name: "test",
  type: "multiOptions",
  default: "a",
  options: [
    {
      name: "B",
      value: "b",
    },
    {
      name: "A",
      value: "a",
    },
    {
      name: "C",
      value: "c",
    },
    {
      name: "D",
      value: "d",
    },
    {
      name: "E",
      value: "e",
    },
  ],
};

const test = {
  displayName: "Events",
  name: "events",
  type: "multiOptions",
  required: true,
  default: [],
  description: "The operation to perform.",
  options: [
    {
      name: "Invoice Generated",
      value: "invoice_generated",
      description:
        'Event triggered when a new invoice is generated. In case of metered billing, this event is triggered when a "Pending" invoice is closed.',
    },
    {
      name: "Date Equal",
      value: "date_equal",
      description: "Field is date. Format: 'YYYY-MM-DD'",
    },
    {
      name: "Invoice Deleted",
      value: "invoice_deleted",
      description: "Event triggered when an invoice is deleted.",
    },
    {
      name: "Subscription Renewal Reminder",
      value: "subscription_renewal_reminder",
      description: "Triggered 3 days before each subscription's renewal.",
    },
    {
      name: "Transaction Created",
      value: "transaction_created",
      description: "Triggered when a transaction is recorded.",
    },
    {
      name: "Transaction Updated",
      value: "transaction_updated",
      description:
        "Triggered when a transaction is updated. E.g. (1) When a transaction is removed, (2) or when an excess payment is applied on an invoice, (3) or when amount_capturable gets updated.",
    },
    {
      name: "Transaction Deleted",
      value: "transaction_deleted",
      description: "Triggered when a transaction is deleted.",
    },
  ],
};
```

‚úÖ Example of **correct** code:

```js
const test = {
  displayName: "Test",
  name: "test",
  type: "multiOptions",
  default: "a",
  options: [
    {
      name: "A",
      value: "a",
    },
    {
      name: "B",
      value: "b",
    },
    {
      name: "C",
      value: "c",
    },
    {
      name: "D",
      value: "d",
    },
    {
      name: "E",
      value: "e",
    },
  ],
};
```

## Links

- [Rule source](../../lib/rules/node-param-multi-options-type-unsorted-items.ts)
- [Test source](../../tests/node-param-multi-options-type-unsorted-items.test.ts)
