[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-missing-final-period

`description` in node parameter must end with a final period if a multiple-sentence description, unless ending with `</code>`. Applicable by extension to `description` in option in options-type and multi-options-type node parameter.

üìã This rule is part of the `plugin:n8n-nodes-base/recommended` config.

üìã This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
  displayName: "Test",
  name: "test",
  type: "string",
  default: "",
  description: "This a sentence. This is another",
};

const test = {
  name: "Username",
  value: "username",
  description: "First sentence. Second sentence",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
  displayName: "Test",
  name: "test",
  type: "string",
  default: "",
  description: "This a sentence. This is another.",
};

const test = {
  displayName: "Test",
  name: "test",
  type: "string",
  default: "",
  description: "This a sentence. This is another</code>",
};

const test = {
  name: "Username",
  value: "username",
  description: "First sentence. Second sentence.",
};

const test = {
  displayName: "Test",
  name: "test",
  type: "string",
  default: "",
  description:
    "Use the multiline editor. Make sure it is in standard PEM key format:-----BEGIN PRIVATE KEY-----KEY DATA GOES HERE-----END PRIVATE KEY-----",
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-missing-final-period.ts)
- [Test source](../../tests/node-param-description-missing-final-period.test.ts)
