[//]: # "File generated from a template. Do not edit this file directly."

# node-param-operation-option-action-miscased

The property `action` in an option in an Operation node parameter must be sentence-cased.

üìã This rule is part of the `plugin:n8n-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
  displayName: "Operation",
  name: "operation",
  type: "options",
  noDataExpression: true,
  displayOptions: {
    show: {
      resource: ["entity"],
    },
  },
  default: "getAll",
  options: [
    {
      name: "Create",
      value: "create",
      description: "Create an entity",
      action: "create an entity",
    },
    {
      name: "Delete",
      value: "delete",
      description: "Delete an entity",
      action: "Delete An entity",
    },
    {
      name: "Get",
      value: "get",
      description: "Retrieve an entity",
      action: "Get an Entity",
    },
    {
      name: "Get All",
      value: "getAll",
      description: "Retrieve all entities",
      action: "Get All Entities",
    },
    {
      name: "Update",
      value: "update",
      description: "Update an entity",
      action: "Update An entity",
    },
  ],
};

const test = {
  displayName: "Action",
  name: "action",
  type: "options",
  noDataExpression: true,
  displayOptions: {
    show: {
      resource: ["entity"],
    },
  },
  default: "getAll",
  options: [
    {
      name: "Create",
      value: "create",
      description: "Create an entity",
      action: "create an entity",
    },
    {
      name: "Delete",
      value: "delete",
      description: "Delete an entity",
      action: "Delete An entity",
    },
    {
      name: "Get",
      value: "get",
      description: "Retrieve an entity",
      action: "Get an Entity",
    },
    {
      name: "Get All",
      value: "getAll",
      description: "Retrieve all entities",
      action: "Get All Entities",
    },
    {
      name: "Update",
      value: "update",
      description: "Update an entity",
      action: "Update An entity",
    },
  ],
};
```

‚úÖ Example of **correct** code:

```js
const test = {
  displayName: "Operation",
  name: "operation",
  type: "options",
  noDataExpression: true,
  displayOptions: {
    show: {
      resource: ["entity"],
    },
  },
  default: "getAll",
  options: [
    {
      name: "Create",
      value: "create",
      description: "Create an entity",
      action: "Create an entity",
    },
    {
      name: "Delete",
      value: "delete",
      description: "Delete an entity",
      action: "Delete an entity",
    },
    {
      name: "Get",
      value: "get",
      description: "Retrieve an entity",
      action: "Get an entity",
    },
    {
      name: "Get All",
      value: "getAll",
      description: "Retrieve all entities",
      action: "Get all entities",
    },
    {
      name: "Update",
      value: "update",
      description: "Update an entity",
      action: "Perform a PATCH request",
    },
  ],
};

const test = {
  displayName: "Action",
  name: "action",
  type: "options",
  noDataExpression: true,
  displayOptions: {
    show: {
      resource: ["entity"],
    },
  },
  default: "getAll",
  options: [
    {
      name: "Create",
      value: "create",
      description: "Create an entity",
      action: "Create an entity",
    },
    {
      name: "Delete",
      value: "delete",
      description: "Delete an entity",
      action: "Delete an entity",
    },
    {
      name: "Get",
      value: "get",
      description: "Retrieve an entity",
      action: "Get an entity",
    },
    {
      name: "Get All",
      value: "getAll",
      description: "Retrieve all entities",
      action: "Get all entities",
    },
    {
      name: "Update",
      value: "update",
      description: "Update an entity",
      action: "Update an entity",
    },
  ],
};
```

## Links

- [Rule source](../../lib/rules/node-param-operation-option-action-miscased.ts)
- [Test source](../../tests/node-param-operation-option-action-miscased.test.ts)
