[//]: # "File generated from a template. Do not edit this file directly."

# node-class-description-outputs-wrong

The number of `outputs` in node class description for any node must be one, or two for If node, or four for Switch node.

üìã This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
class TestNode {
  description = {
    displayName: "Test",
    name: "test",
    icon: "file:test.svg",
    group: ["transform"],
    version: 1,
    subtitle: '={{ $parameter["operation"] + ": " + $parameter["resource"] }}',
    description: "This is a sentence",
    defaults: {
      name: "Test",
    },
    inputs: ["main"],
    outputs: [],
  };
}
```

‚úÖ Example of **correct** code:

```js
class TestNode {
  description = {
    displayName: "Test",
    name: "test",
    icon: "file:test.svg",
    group: ["transform"],
    version: 1,
    subtitle: '={{ $parameter["operation"] + ": " + $parameter["resource"] }}',
    description: "This is a sentence",
    defaults: {
      name: "Test",
    },
    inputs: ["main"],
    outputs: ["main"],
  };
}
```

## Links

- [Rule source](../../lib/rules/node-class-description-outputs-wrong.ts)
- [Test source](../../tests/node-class-description-outputs-wrong.test.ts)
