[//]: # "File generated from a template. Do not edit this file directly."

# node-param-resource-with-plural-option

Option `name` for a Resource node parameter must be singular.

üìã This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
  displayName: "Resource",
  name: "resource",
  type: "options",
  noDataExpression: true,
  options: [
    {
      name: "Users",
      value: "user",
    },
  ],
  default: "contact",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
  displayName: "Resource",
  name: "resource",
  type: "options",
  noDataExpression: true,
  options: [
    {
      name: "User",
      value: "user",
    },
  ],
  default: "contact",
};
```

## Links

- [Rule source](../../lib/rules/node-param-resource-with-plural-option.ts)
- [Test source](../../tests/node-param-resource-with-plural-option.test.ts)
