[//]: # "File generated from a template. Do not edit this file directly."

# node-param-display-name-wrong-for-simplify

`displayName` for Simplify node parameter must be Simplify

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/recommended` config.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

ğŸ”§ Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

âŒ Example of **incorrect** code:

```js
const test = {
  displayName: "SimplifyResponse",
  name: "simple",
  type: "boolean",
  default: true,
};
```

âœ… Example of **correct** code:

```js
const test = {
  displayName: "Simplify",
  name: "simple",
  type: "boolean",
  default: true,
};
```

## Links

- [Rule source](../../lib/rules/node-param-display-name-wrong-for-simplify.ts)
- [Test source](../../tests/node-param-display-name-wrong-for-simplify.test.ts)
