[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-identical-to-display-name

`description` in node parameter must not be identical to `displayName`.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/recommended` config.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

ğŸ”§ Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

âŒ Example of **incorrect** code:

```js
const test = {
  displayName: "Test",
  name: "test",
  type: "string",
  default: "",
  description: "The test",
};
```

âœ… Example of **correct** code:

```js
const test = {
  displayName: "Test",
  name: "test",
  type: "string",
  default: "",
  description: "This is a test",
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-identical-to-display-name.ts)
- [Test source](../../tests/node-param-description-identical-to-display-name.test.ts)
