[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-miscased-url

`URL` in `description` in node parameter must be fully uppercased. Applicable by extension to `description` in option in options-type and multi-options-type node parameter.

üìã This rule is part of the `plugin:n8n-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	displayName: "Operation",
	name: "operation",
	type: "string",
	default: "",
	description: "The url of the curly user",
};

const test = {
	displayName: "Operation",
	name: "operation",
	type: "string",
	default: "",
	description: "An avatar image url. note: the image url needs to be https.",
};

const test = {
	displayName: "Operation",
	name: "operation",
	type: "string",
	default: "",
	description: "The Url of the curly user",
};

const test = {
	displayName: "Operation",
	name: "operation",
	type: "string",
	default: "",
	description: "Url of the user",
};

const test = {
	name: "User ID",
	value: "userId",
	description: "The url of the curly user",
};

const test = {
	name: "User ID",
	value: "userId",
	description: "The Url of the curly user",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	displayName: "Operation",
	name: "operation",
	type: "string",
	default: "",
	description: "The URL of the user",
};

const test = {
	name: "User ID",
	value: "userId",
	description: "The URL of the user",
};

const test = {
	name: "User ID",
	value: "userId",
	description: "The cURL command",
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-miscased-url.ts)
- [Test source](../../tests/node-param-description-miscased-url.test.ts)
