[//]: # "File generated from a template. Do not edit this file directly."

# node-param-type-options-missing-from-limit

`typeOptions` in Limit node parameter must be present.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/recommended` config.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

ğŸ”§ Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

âŒ Example of **incorrect** code:

```js
const test = {
  displayName: "Limit",
  name: "limit",
  default: 50,
  type: "number",
};
```

âœ… Example of **correct** code:

```js
const test = {
  displayName: "Limit",
  name: "limit",
  default: 50,
  type: "number",
  typeOptions: {
    minValue: 1,
  },
};
```

## Links

- [Rule source](../../lib/rules/node-param-type-options-missing-from-limit.ts)
- [Test source](../../tests/node-param-type-options-missing-from-limit.test.ts)
