[//]: # "File generated from a template. Do not edit this file directly."

# node-execute-block-double-assertion-for-items

In the `execute()` method there is no need to double assert the type of `items.length`.

üìã This rule is part of the `plugin:n8n-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
class TestNode {
    async execute() {
        const items = this.getInputData();
        const length = items.length as unknown as number;
    }
}

class TestNode {
    async execute() {
        const items = this.getInputData();
        const length = (items.length as unknown) as number;
    }
}
```

‚úÖ Example of **correct** code:

```js
class TestNode {
  async execute() {
    const items = this.getInputData();
    const length = items.length;
  }
}
```

## Links

- [Rule source](../../lib/rules/node-execute-block-double-assertion-for-items.ts)
- [Test source](../../tests/node-execute-block-double-assertion-for-items.test.ts)
