[//]: # "File generated from a template. Do not edit this file directly."

# node-param-default-wrong-for-simplify

`default` for a Simplify node parameter must be `true`.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/nodes` config.

ğŸ”§ Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

âŒ Example of **incorrect** code:

```js
const test = {
  displayName: "Simplify",
  name: "simple",
  type: "boolean",
  default: false,
};
```

âœ… Example of **correct** code:

```js
const test = {
  displayName: "Simplify",
  name: "simpl",
  type: "boolean",
  default: true,
};
```

## Links

- [Rule source](../../lib/rules/node-param-default-wrong-for-simplify.ts)
- [Test source](../../tests/node-param-default-wrong-for-simplify.test.ts)
