[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-wrong-for-dynamic-options

`description` in dynamic-options-type node parameter must be `Choose from the list, or specify an ID using an <a href="https://docs.n8n.io/nodes/expressions.html#expressions">expression</a>`

üìã This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
  displayName: "Field Name or ID",
  name: "field",
  type: "options",
  typeOptions: {
    loadOptionsMethod: "getFields",
  },
  default: "",
  description: "This is a sentence",
};

const test = {
  displayName: "Field Name or ID",
  name: "field",
  type: "options",
  typeOptions: {
    loadOptionsMethod: "getFields",
  },
  default: "",
  description: "This is a sentence. This is another.",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
  displayName: "Field Name or ID",
  name: "field",
  type: "options",
  typeOptions: {
    loadOptionsMethod: "getFields",
  },
  default: "",
  description:
    'Choose from the list, or specify an ID using an <a href="https://docs.n8n.io/nodes/expressions.html#expressions">expression</a>',
};

const test = {
  displayName: "Field Name or ID",
  name: "field",
  type: "options",
  typeOptions: {
    loadOptionsMethod: "getFields",
  },
  default: "",
  description:
    'This is a sentence. Choose from the list, or specify an ID using an <a href="https://docs.n8n.io/nodes/expressions.html#expressions">expression</a>.',
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-wrong-for-dynamic-options.ts)
- [Test source](../../tests/node-param-description-wrong-for-dynamic-options.test.ts)
