[//]: # "File generated from a template. Do not edit this file directly."

# node-param-display-name-miscased

`displayName` in node parameter or in fixed collection section must title cased. Applicable by extension to `name` in options-type or multi-options-type node parameter.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/recommended` config.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

ğŸ”§ Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

âŒ Example of **incorrect** code:

```js
const test = {
  displayName: "Test of tests",
  name: "testOfTests",
  type: "string",
  default: "",
};

const test = {
  name: "Test of tests",
  value: "testOfTests",
};

const test = {
  displayName: "Test",
  name: "test",
  type: "fixedCollection",
  default: "a",
  options: [
    {
      displayName: "Details test",
      name: "detailsTest",
      values: [
        {
          displayName: "A",
          name: "a",
          type: "string",
          default: "",
        },
      ],
    },
  ],
};
```

âœ… Example of **correct** code:

```js
const test = {
  displayName: "Test of Tests",
  name: "testOfTests",
  type: "string",
  default: "",
};

const test = {
  name: "Test of Tests",
  value: "testOfTests",
};
```

## Links

- [Rule source](../../lib/rules/node-param-display-name-miscased.ts)
- [Test source](../../tests/node-param-display-name-miscased.test.ts)
