[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-wrong-for-limit

`description` for Limit node parameter must be `Max number of results to return`

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

ğŸ”§ Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

âŒ Example of **incorrect** code:

```js
const test = {
  displayName: "Limit",
  name: "limit",
  type: "number",
  default: "",
  description: "Wrong",
};
```

âœ… Example of **correct** code:

```js
const test = {
  displayName: "Limit",
  name: "limit",
  type: "number",
  default: "",
  description: "Max number of results to return",
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-wrong-for-limit.ts)
- [Test source](../../tests/node-param-description-wrong-for-limit.test.ts)
