[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-missing-from-dynamic-options

`description` in dynamic-options-type node parameter must be present.

üìã This rule is part of the `plugin:n8n-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	displayName: "Field Name or ID",
	name: "field",
	type: "options",
	typeOptions: {
		loadOptionsMethod: "getFields",
	},
	default: "",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	displayName: "Field Name or ID",
	name: "field",
	type: "options",
	description:
		'Choose from the list, or specify an ID using an <a href="https://docs.n8n.io/code/expressions/">expression</a>',
	typeOptions: {
		loadOptionsMethod: "getFields",
	},
	default: "",
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-missing-from-dynamic-options.ts)
- [Test source](../../tests/node-param-description-missing-from-dynamic-options.test.ts)
