[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-wrong-for-return-all

`description` for Return All node parameter must be `Whether to return all results or only up to a given limit`

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/recommended` config.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

ğŸ”§ Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

âŒ Example of **incorrect** code:

```js
const test = {
  displayName: "Return All",
  name: "returnAll",
  type: "boolean",
  default: true,
  description: "Wrong",
};
```

âœ… Example of **correct** code:

```js
const test = {
  displayName: "Return All",
  name: "returnAll",
  type: "boolean",
  default: true,
  description: "Whether to return all results or only up to a given limit",
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-wrong-for-return-all.ts)
- [Test source](../../tests/node-param-description-wrong-for-return-all.test.ts)
