[//]: # "File generated from a template. Do not edit this file directly."

# node-class-description-icon-not-svg

`icon` in node class description should be an SVG icon.

üìã This rule is part of the `plugin:n8n-nodes-base/non-autofixable` config.

## Examples

‚ùå Example of **incorrect** code:

```js
class TestNode {
  description = {
    displayName: "Test",
    name: "testTrigger",
    icon: "file:test.png",
    group: ["transform"],
    version: 1,
    subtitle: '={{ $parameter["operation"] + ": " + $parameter["resource"] }}',
    description: "This is a sentence",
    defaults: {
      name: "Test",
    },
    inputs: ["main"],
    outputs: ["main"],
  };
}
```

‚úÖ Example of **correct** code:

```js
class TestNode {
  description = {
    displayName: "TestTrigger",
    name: "testTrigger",
    icon: "file:test.svg",
    group: ["transform"],
    version: 1,
    subtitle: '={{ $parameter["operation"] + ": " + $parameter["resource"] }}',
    description: "This is a sentence",
    defaults: {
      name: "Test",
    },
    inputs: ["main"],
    outputs: ["main"],
  };
}
```

## Links

- [Rule source](../../lib/rules/node-class-description-icon-not-svg.ts)
- [Test source](../../tests/node-class-description-icon-not-svg.test.ts)
