[//]: # "File generated from a template. Do not edit this file directly."

# node-execute-block-missing-continue-on-fail

The `execute()` method in a node must implement `continueOnFail` in a try-catch block.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/recommended` config.

ğŸ“‹ This rule is part of the `plugin:n8n-nodes-base/non-autofixable` config.

## Examples

âŒ Example of **incorrect** code:

```js
class TestNode {
  description = {
    displayName: "Test",
    name: "test",
    icon: "file:test.svg",
    group: ["transform"],
    version: 1,
    subtitle: '={{ $parameter["operation"] + ": " + $parameter["resource"] }}',
    description: "This is a sentence",
    defaults: {
      name: "Test",
    },
    inputs: ["main"],
    outputs: ["main"],
  };
  async execute() {
    for (let i = 0; i < items.length; i++) {
      try {
        // ...
      } catch (error) {
        // ...
      }
    }
  }
}

class TestNode {
  description = {
    displayName: "Test",
    name: "test",
    icon: "file:test.svg",
    group: ["transform"],
    version: 1,
    subtitle: '={{ $parameter["operation"] + ": " + $parameter["resource"] }}',
    description: "This is a sentence",
    defaults: {
      name: "Test",
    },
    inputs: ["main"],
    outputs: ["main"],
  };
  async execute() {
    for (let i = 0; i < items.length; i++) {
      // ...
    }
  }
}
```

âœ… Example of **correct** code:

```js
class TestNode {
  description = {
    displayName: "Test",
    name: "test",
    icon: "file:test.svg",
    group: ["transform"],
    version: 1,
    subtitle: '={{ $parameter["operation"] + ": " + $parameter["resource"] }}',
    description: "This is a sentence",
    defaults: {
      name: "Test",
    },
    inputs: ["main"],
    outputs: ["main"],
  };
  async execute() {
    for (let i = 0; i < items.length; i++) {
      try {
        // ...
      } catch (error) {
        if (this.continueOnFail()) {
          // ...
        }
      }
    }
  }
}
```

## Links

- [Rule source](../../lib/rules/node-execute-block-missing-continue-on-fail.ts)
- [Test source](../../tests/node-execute-block-missing-continue-on-fail.test.ts)
