[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-empty-string

`description` in node parameter or in option in options-type and multi-options-type param must be filled out or removed. Applicable by extension to `description` in option in options-type and multi-options-type node parameter.

üìã This rule is part of the `plugin:n8n-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
  displayName: "Test",
  name: "test",
  type: "string",
  default: "",
  description: "",
};

const test = {
  name: "User ID",
  value: "userId",
  description: "",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
  displayName: "Test",
  name: "test",
  type: "string",
  default: "",
};

const test = {
  name: "User ID",
  value: "userId",
  description: "The ID of the user",
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-empty-string.ts)
- [Test source](../../tests/node-param-description-empty-string.test.ts)
