[//]: # "File generated from a template. Do not edit this file directly."

# node-param-options-type-unsorted-items

Items in options-type node parameter must be alphabetized by `name` if five or more than five.

üìã This rule is part of the `plugin:n8n-nodes-base/nodes` config.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "options",
	default: "a",
	options: [
		{
			name: "B",
			value: "b",
		},
		{
			name: "A",
			value: "a",
		},
		{
			name: "C",
			value: "c",
		},
		{
			name: "D",
			value: "d",
		},
		{
			name: "E",
			value: "e",
		},
	],
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	displayName: "Test",
	name: "test",
	type: "options",
	default: "a",
	options: [
		{
			name: "A",
			value: "a",
		},
		{
			name: "B",
			value: "b",
		},
		{
			name: "C",
			value: "c",
		},
		{
			name: "D",
			value: "d",
		},
		{
			name: "E",
			value: "e",
		},
	],
};
```

## Links

- [Rule source](../../lib/rules/node-param-options-type-unsorted-items.ts)
- [Test source](../../tests/node-param-options-type-unsorted-items.test.ts)
