[//]: # "File generated from a template. Do not edit this file directly."

# node-param-option-name-containing-star

Option `name` in options-type node parameter must not contain `*`. Use `[All]` instead.

üìã This rule is part of the `plugin:n8n-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
  displayName: "Test",
  name: "test",
  type: "options",
  default: "some",
  options: [
    {
      name: "*",
      value: "all",
    },
    {
      name: "Some",
      value: "some",
    },
  ],
};
```

‚úÖ Example of **correct** code:

```js
const test = {
  displayName: "Test",
  name: "test",
  type: "options",
  default: "some",
  options: [
    {
      name: "[All]",
      value: "all",
    },
    {
      name: "Some",
      value: "some",
    },
  ],
};
```

## Links

- [Rule source](../../lib/rules/node-param-option-name-containing-star.ts)
- [Test source](../../tests/node-param-option-name-containing-star.test.ts)
